runtime/src/bank/partitioned_epoch_rewards/distribution.rs

```diff
@@ -150,17 +169,49 @@ impl Bank {
         })
     }
 
-    /// store stake rewards in partition
-    /// return the sum of all the stored rewards
+    /// Store stake rewards in partition
+    /// Returns DistributionStorageResults containing the sum of all the rewards
+    /// stored, the sum of all rewards burned, and the updated StakeRewards.
+    /// Because stake accounts are checked in calculation, and further state
+    /// mutation prevents by stake-program restrictions, there should never be
+    /// rewards burned.
     ///
-    /// Note: even if staker's reward is 0, the stake account still needs to be stored because
-    /// credits observed has changed
-    fn store_stake_accounts_in_partition(&self, stake_rewards: &[StakeReward]) -> u64 {
-        self.store_accounts((self.slot(), stake_rewards));
-        stake_rewards
-            .iter()
-            .map(|stake_reward| stake_reward.stake_reward_info.lamports)
-            .sum::<i64>() as u64
+    /// Note: even if staker's reward is 0, the stake account still needs to be
+    /// stored because credits observed has changed
+    fn store_stake_accounts_in_partition(
+        &self,
+        stake_rewards: PartitionedStakeRewards,
+    ) -> DistributionStorageResults {
+        let mut lamports_distributed = 0;
+        let mut lamports_burned = 0;
+        let mut updated_stake_rewards = vec![];
+        let stakes_cache = self.stakes_cache.stakes();
+        let stakes_cache_accounts = stakes_cache.stake_delegations();
+        for partitioned_stake_reward in stake_rewards.into_iter() {
+            let stake_pubkey = partitioned_stake_reward.stake_pubkey;
+            let reward_amount = partitioned_stake_reward.stake_reward_info.lamports as u64;
+            match Self::build_updated_stake_reward(stakes_cache_accounts, partitioned_stake_reward)
+            {
+                Ok(stake_reward) => {
+                    lamports_distributed += reward_amount;
+                    updated_stake_rewards.push(stake_reward);
+                }
+                Err(err) => {
+                    debug!(
+                        "bank::distribution::store_stake_accounts_in_partition() failed for {}: {:?}",
+                        stake_pubkey, err
+                    );
```

Comment:
What do you think about bumping this up to an `error` since it shouldn't ever happen?

Alternatively, is it worth asserting and not including the `lamports_burned` field at all? Is there a case where this would ever get tripped?

Reply:
> What do you think about bumping this up to an error since it shouldn't ever happen?

Sure, I'll lift this.

> Alternatively, is it worth asserting and not including the lamports_burned field at all? Is there a case where this would ever get tripped?

As I said in the description, this *shouldn't* ever happen. But I ultimately decided to mirror the existing process as closely as possible, and since that involves dropping rewards that that error in `stake_stake::redeem_rewards()` instead of panicking, I figured it made sense to log any incidences of this. But if you feel strongly, we can switch to asserting.
Alternately, we could wait until after some amount of testing (potentially even on testnet) and then switch to asserting.